### PR TITLE
Upgrade vertaktoid to android 9

### PR DESCRIPTION
**Pull request by roewenstrunk**
The code of vertaktoid is now compatible with Android SDK 28.
Libraries were upgraded.
App permissions now granted individually at run-time, not all-or-nothing at install time. Feature of Android 6.

### CODE REVIEW

**Comment by roewenstrunk** [comment]
Please try the vertaktoid release candidate published at https://github.com/cemfi/vertaktoid/releases/tag/3.0.0-rc and give feedback to this pull request

File: build.gradle [hunk 1]
```diff
@@ -8,6 +8,36 @@ def keystorePropertiesFile = rootProject.file("keystore.properties")
 def keystoreProperties = new Properties()
 
 // Load your keystore.properties file into the keystoreProperties object.
+
```
Comment (bwbohl) [inline build.gradle]: is there any sense in adding theses empty lines?

**Comment by bwbohl** [comment]
Trying to build With Android Studio 2021.1.1 Patch 1 recommends to Upgrade Gradle

**Comment by bwbohl** [comment]
Ok after installing a debug build I get the message "Please install a Fie Manager" when tapping the folder icon in Vertaktoid in order to open a folder containing source images.
This is strange as Samsung “My Files” is installed on the Samsung S7+ Tablet by default. Running Android 11.

**Comment by bwbohl** [comment]
Even after installing Google Files the Problem persists

**Comment by bwbohl** [comment]
When I try opening following IIIF manifest: 
https://digital.blb-karlsruhe.de/i3f/v20/6146583/manifest

the application fails:
```
D/AndroidRuntime: Shutting down VM
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: zemfi.de.vertaktoid, PID: 28337
    java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.res.Resources android.content.Context.getResources()' on a null object reference
        at android.content.ContextWrapper.getResources(ContextWrapper.java:101)
        at android.view.ContextThemeWrapper.getResourcesInternal(ContextThemeWrapper.java:134)
        at android.view.ContextThemeWrapper.getResources(ContextThemeWrapper.java:128)
        at android.support.v7.app.AppCompatActivity.getResources(AppCompatActivity.java:543)
        at android.widget.ToastPresenter.<init>(ToastPresenter.java:121)
        at android.widget.Toast$TN.<init>(Toast.java:1151)
        at android.widget.Toast.<init>(Toast.java:259)
        at android.widget.Toast.makeText(Toast.java:891)
        at android.widget.Toast.makeText(Toast.java:879)
        at zemfi.de.vertaktoid.MainActivity.actionOpen(MainActivity.java:355)
        at zemfi.de.vertaktoid.IiifManifest.chooseDownloadDirectory(IiifManifest.java:181)
        at zemfi.de.vertaktoid.IiifManifest$3.onResponse(IiifManifest.java:148)
        at zemfi.de.vertaktoid.IiifManifest$3.onResponse(IiifManifest.java:96)
        at com.android.volley.toolbox.JsonRequest.deliverResponse(JsonRequest.java:100)
        at com.android.volley.ExecutorDelivery$ResponseDeliveryRunnable.run(ExecutorDelivery.java:102)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:246)
        at android.app.ActivityThread.main(ActivityThread.java:8512)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:596)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)
```

So consequently I fail both on local files and IIIF manifest files.
Any help for fix appreciated ;-)

**Comment by bwbohl** [comment]
some details:
`onActivityResult() called with: requestCode = [0], resultCode = [0], data = [null]`

**Comment by bwbohl** [comment]
After Updating to Android 12 the File Manager Call work and local image files can be opened. Downloading the above IIIF.manifest still fails. though.

**Comment by bwbohl** [comment]
when marking measures the facsimile disappears and the message no folder selected appears when flipping the tablet from portrait to landscape view or vice versa.

**Comment by Hizkie** [comment]
> https://digital.blb-karlsruhe.de/i3f/v20/6146583/manifest

**Comment by Hizkie** [comment]
> when marking measures the facsimile disappears and the message no folder selected appears when flipping the tablet from portrait to landscape view or vice versa.

**Comment by hizclick** [comment]
> https://digital.blb-karlsruhe.de/i3f/v20/6146583/manifest



> When I try opening following IIIF manifest: https://digital.blb-karlsruhe.de/i3f/v20/6146583/manifest
> 
> the application fails:
> 
> ```
> D/AndroidRuntime: Shutting down VM
> E/AndroidRuntime: FATAL EXCEPTION: main
>     Process: zemfi.de.vertaktoid, PID: 28337
>     java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.res.Resources android.content.Context.getResources()' on a null object reference
>         at android.content.ContextWrapper.getResources(ContextWrapper.java:101)
>         at android.view.ContextThemeWrapper.getResourcesInternal(ContextThemeWrapper.java:134)
>         at android.view.ContextThemeWrapper.getResources(ContextThemeWrapper.java:128)
>         at android.support.v7.app.AppCompatActivity.getResources(AppCompatActivity.java:543)
>         at android.widget.ToastPresenter.<init>(ToastPresenter.java:121)
>         at android.widget.Toast$TN.<init>(Toast.java:1151)
>         at android.widget.Toast.<init>(Toast.java:259)
>         at android.widget.Toast.makeText(Toast.java:891)
>         at android.widget.Toast.makeText(Toast.java:879)
>         at zemfi.de.vertaktoid.MainActivity.actionOpen(MainActivity.java:355)
>         at zemfi.de.vertaktoid.IiifManifest.chooseDownloadDirectory(IiifManifest.java:181)
>         at zemfi.de.vertaktoid.IiifManifest$3.onResponse(IiifManifest.java:148)
>         at zemfi.de.vertaktoid.IiifManifest$3.onResponse(IiifManifest.java:96)
>         at com.android.volley.toolbox.JsonRequest.deliverResponse(JsonRequest.java:100)
>         at com.android.volley.ExecutorDelivery$ResponseDeliveryRunnable.run(ExecutorDelivery.java:102)
>         at android.os.Handler.handleCallback(Handler.java:938)
>         at android.os.Handler.dispatchMessage(Handler.java:99)
>         at android.os.Looper.loop(Looper.java:246)
>         at android.app.ActivityThread.main(ActivityThread.java:8512)
>         at java.lang.reflect.Method.invoke(Native Method)
>         at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:596)
>         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)
> ```
> 
> So consequently I fail both on local files and IIIF manifest files. Any help for fix appreciated ;-)

Because of permission, can you try to download the file in one of the android's standard directories?

**Pull request merged**